mne/io/egi/egimff.py

```diff
@@ -462,10 +469,22 @@ def __init__(self, input_fname, eog=None, misc=None,
         info['utc_offset'] = egi_info['utc_offset']
         info['device_info'] = dict(type=egi_info['device'])
 
-        # First: EEG
-        ch_names = [channel_naming % (i + 1) for i in
-                    range(egi_info['n_channels'])]
+        # read in the montage, if it exists
+        ch_names, mon = _read_locs(input_fname, egi_info, channel_naming)
 
+        # First: EEG
+        _auto_ch_names = [channel_naming % (i + 1) for i in
+                          range(egi_info['n_channels'])]
+        if ch_names is None:
+            ch_names = _auto_ch_names
+        else:
+            missing_ch_name_mask = [len(name) == 0 for name in ch_names]
+            if any(missing_ch_name_mask):
+                logger.info(
+                    'some channel names were missing from  coordinates.xml; '
+                    'falling back to auto-naming for those channels')
+            ch_names = np.where([len(name) == 0 for name in ch_names],
+                                _auto_ch_names, ch_names).tolist()
```

Comment:
I wonder if this is even necessary anymore, now that `_read_locs` uses the `number` field automatically if `name` is empty or missing?